core/src/main/java/io/questdb/std/IntList.java

```diff
@@ -34,39 +34,41 @@
 public class IntList implements Mutable, Sinkable {
     private static final int DEFAULT_ARRAY_SIZE = 16;
     private static final int NO_ENTRY_VALUE = -1;
-    private int[] buffer;
+    private final int initialCapacity;
+    private int[] data;
     private int pos = 0;
 
     public IntList() {
         this(DEFAULT_ARRAY_SIZE);
     }
 
     public IntList(int capacity) {
-        this.buffer = new int[Math.max(capacity, DEFAULT_ARRAY_SIZE)];
+        this.initialCapacity = Math.max(capacity, DEFAULT_ARRAY_SIZE);
```

Comment:
Why we cannot create an IntList of capacity 2 or 4 anymore? Some usages know in advance how many elements will be used and it can be below 16.

Reply:
That's a valid point, especially considering we don't have such logic in `LongList`. Changed this in 269bf18b8b6f0038da1d2003efd1fc15134fd28d